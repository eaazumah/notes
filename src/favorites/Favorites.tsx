import React from 'react';
import {renderCardItem} from '../@shared/components/Card';
import {renderEmpty} from '../@shared/components/Empty';
import Header from '../@shared/components/Header';
import useNotes from '../@shared/hooks/useNotes';
import Container from '../@shared/styled/Container';
import FlatList from '../@shared/styled/FlatList';

interface Props {}

const Favorites: React.FC<Props> = () => {
  const {favorites} = useNotes();
  return (
    <Container>
      <Header title={'Favorites'} />
      <FlatList
        data={favorites}
        renderItem={renderCardItem}
        ListEmptyComponent={renderEmpty}
        scrollEnabled={!!favorites.length}
      />
    </Container>
  );
};

export default Favorites;
