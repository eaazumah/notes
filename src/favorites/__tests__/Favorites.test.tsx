import {cleanup, render} from '@testing-library/react-native';
import React from 'react';
import TestProvider from '../../__testUtils__/Providers';
import Favorites from '../Favorites';

describe('Favorites', () => {
  afterEach(cleanup);

  test('Favorites renders correctly', async () => {
    const screen = render(
      <TestProvider>
        <Favorites />
      </TestProvider>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });

  test('Favorites functionality test', async () => {
    const screen = render(
      <TestProvider>
        <Favorites />
      </TestProvider>,
    );
    const headerTitle = await screen.findByText('Favorites');
    expect(headerTitle).toBeTruthy();
  });
});
