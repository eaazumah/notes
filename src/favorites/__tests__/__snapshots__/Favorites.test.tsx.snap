// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites Favorites renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5F5F5",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "marginTop": 50,
          "paddingHorizontal": 17,
          "width": "100%",
        },
      ]
    }
    testID="header"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#242A37",
            "fontSize": 30,
            "fontWeight": "800",
          },
        ]
      }
      testID="title"
    >
      Favorites
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="open-settings"
    >
      <Icon
        color="#242A37"
        name="settings"
        size={25}
        testID="icon"
      />
    </View>
  </View>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={false}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "backgroundColor": "#F5F5F5",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "width": "100%",
        },
      ]
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5F5F5",
              "height": 1134,
              "justifyContent": "center",
            },
          ]
        }
      >
        <SvgMock
          height={200}
          width={200}
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
